tools/state-viewer/src/commands.rs

```diff
@@ -567,32 +578,45 @@ pub(crate) fn print_chain(
                 {
                     cur_epoch_id = Some(epoch_id.clone());
-                    if epoch_manager.is_next_block_epoch_start(header.prev_hash()).unwrap() {
-                        println!("{:?}", account_id_to_blocks);
-                        account_id_to_blocks = HashMap::new();
-                        println!(
-                            "Epoch {} Validators {:?}",
-                            format_hash(epoch_id.0, show_full_hashes),
-                            epoch_manager
-                                .get_epoch_block_producers_ordered(&epoch_id, header.hash())
-                                .unwrap()
-                        );
-                    }
-                    let block_producer =
-                        epoch_manager.get_block_producer(&epoch_id, header.height()).unwrap();
+                    match epoch_manager.is_next_block_epoch_start(header.prev_hash()) {
+                        Ok(true) => {
+                            println!("{:?}", account_id_to_blocks);
+                            account_id_to_blocks = HashMap::new();
+                            println!(
+                                "Epoch {} Validators {:?}",
+                                format_hash(epoch_id.0, show_full_hashes),
+                                epoch_manager
+                                    .get_epoch_block_producers_ordered(&epoch_id, header.hash())
+                            );
+                        }
+                        Err(err) => {
+                            println!("Don't know if next block is epoch start: {err:?}");
+                        }
+                        _ => {}
+                    };
+                    let block_producer = epoch_manager
+                        .get_block_producer(&epoch_id, header.height())
+                        .map(|account_id| account_id.to_string())
+                        .ok()
+                        .unwrap_or("error".to_owned());
                     account_id_to_blocks
                         .entry(block_producer.clone())
                         .and_modify(|e| *e += 1)
                         .or_insert(1);
 
-                    let block = chain_store.get_block(&block_hash).unwrap().clone();
+                    let block = if let Ok(block) = chain_store.get_block(&block_hash) {
+                        block.clone()
+                    } else {
+                        continue;
+                    };
 
                     let mut chunk_debug_str: Vec<String> = Vec::new();
 
                     for shard_id in 0..header.chunk_mask().len() {
                         let chunk_producer = epoch_manager
                             .get_chunk_producer(&epoch_id, header.height(), shard_id as u64)
-                            .unwrap();
+                            .map(|account_id| account_id.to_string())
+                            .unwrap_or("error".to_owned());
```

Comment:
Maybe be a bit more specific? Unless it's going to be clear from context that this error is about chunk_producer.

Reply:
Done